instrumentation/grape/README.md

```diff
@@ -0,0 +1,90 @@
+# OpenTelemetry Grape Instrumentation
+
+The Grape instrumentation is a community-maintained instrumentation for [Grape][grape], a REST-like API framework for Ruby.
+
+It relies on the Grape built-in support for `ActiveSupport::Notifications` (more info [here](https://github.com/ruby-grape/grape#active-support-instrumentation)).
+
+It currently supports the following events:
+
+- `endpoint_run.grape`
+- `endpoint_render.grape`
+- `endpoint_run_filters.grape`
+- `format_response.grape`
+
+## How do I get started?
+
+Install the gem using:
+
+```
+gem install opentelemetry-instrumentation-grape
+```
+
+Or, if you use [bundler][bundler-home], include `opentelemetry-instrumentation-grape` in your `Gemfile`.
+
+## Usage
+
+To use the instrumentation, call `use` with the name of the instrumentation:
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  c.use 'OpenTelemetry::Instrumentation::Grape'
+end
+```
+
+Since Grape is "designed to run on Rack or complement existing web application frameworks such as Rails and Sinatra", it is recommended to use this instrumentation along with the Rack, Rails and/or Sinatra instrumentations.
+
+Alternatively, you can also call `use_all` to install all the available instrumentation.
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  c.use_all
+end
+```
+
+### Configuration options
+
+#### `:ignored_events` (array)
+
+Indicate if any events should not produce spans.
+
+- Accepted values: `:endpoint_render`, `:endpoint_run_filters`, `:format_response`.
+- Defaults to `[]` (no ignored events).
+
+Example:
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  c.use 'OpenTelemetry::Instrumentation::Grape', { ignored_events: [:endpoint_run_filters] }
+end
+```
+
+Note that the `endpoint_run` event cannot be ignored. If you need to disable the instrumentation, set `:enabled` to `false`:
```

Comment:
We decided to exclude `endpoint_run` from the "ignorable" events because, since it's the "main" span, IMO it wouldn't make a lot of sense to ignore it. It's the also span that has all the more informative attributes, and also if it's excluded all the other spans would be orphaned (at least as far as Grape is concerned, because they would still grouped together into the trace by the middleware span).

Reply:
I decided to keep the exclusion of `endpoint_run` from the "ignorable" events with the new changes too because it's the parent event, where the Rack span name is modified and the most important attributes are added.